core/net/ipv6/uip-nd6.c

```diff
@@ -489,23 +489,32 @@ na_input(void)
     PRINTF("NA received is bad\n");
     goto discard;
   } else {
-    uip_lladdr_t *lladdr;
+    uip_lladdr_t lladdr_aligned;
     nbr = uip_ds6_nbr_lookup(&UIP_ND6_NA_BUF->tgtipaddr);
-    lladdr = (uip_lladdr_t *)uip_ds6_nbr_get_ll(nbr);
     if(nbr == NULL) {
       goto discard;
     }
+    extract_lladdr_aligned(&lladdr_aligned);
+
     if(nd6_opt_llao != 0) {
       is_llchange =
-        memcmp(&nd6_opt_llao[UIP_ND6_OPT_DATA_OFFSET], (void *)lladdr,
+        memcmp(&nd6_opt_llao[UIP_ND6_OPT_DATA_OFFSET], (void *)&lladdr_aligned,
                UIP_LLADDR_LEN);
```

Comment:
With the latest update, lladdr_aligned will always set to &nd6_opt_llao[UIP_ND6_OPT_DATA_OFFSET] and not NBR lladdr so the compare will always say no change.